### ISSUE TITLE
Build site in CI

### ISSUE THREAD

**Issue by bgilbert** [coreos/fedora-coreos-docs#357]
PR CI should run `./build.sh`, upload the artifacts somewhere browsable, and post a link in the PR.

**Comment by RishiKumarRay** [comment]
I would like to try this

**Comment by bgilbert** [comment]
Great!  For security, GitHub recommends splitting privileged jobs like this into two pieces, similar to [this](https://github.com/coreos/fedora-coreos-streams/blob/main/.github/workflows/print-rollouts.yml) and [this](https://github.com/coreos/fedora-coreos-streams/blob/main/.github/workflows/post-rollouts.yml).

@dustymabe, is there a good place we could upload staging content?  Maybe we could create an S3 bucket and a token with write access to it?

**Comment by travier** [comment]
We could enable GitHub pages here and push to a docs branch with a subpath based on PR number?

**Comment by travier** [comment]
It's also fairly easy to setup GitHub Pages for a forked repo: https://github.com/coreos/fedora-coreos-tracker/blob/main/docs/testing-project-documentation-changes.md

**Comment by travier** [comment]
> I'd rather not fill the production repo with a bunch of test builds. If we want to go the GitHub Pages route, maybe we could create a dedicated repo for that. Even then, it seems a bit awkward for CI runs to live in long-term Git history.

I agree that having a dedicated testing repo would make sense and would simplify the permission setup while keeping the production repo clean. We could do a periodic cleanup of old branches but I don't think it really matters.
